File: backend/src/main/java/com/pravdinm/synthese/model/delivery/Order.java
package com.pravdinm.synthese.model.delivery;

import com.pravdinm.synthese.model.user.Client;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@Document(collection = "order")
public class Order {

    @Id
    private String orderId;

    @Builder.Default
    private List<Listing> listingList = new ArrayList<>();

    @Field
    private String address;

    @Field
    private String city;

    @Field
    private float cost;

    @Field
    private String orderInfo;

    @CreatedDate
    @Builder.Default
    private Date creationDate = new Date();

    @Builder.Default
    private Boolean isAccepted = false;

    @Builder.Default
    private Date shippingDate = new Date();

    @DBRef
    private Client client;

    public Order() {
        super();
        this.listingList = new ArrayList<>();
        this.creationDate = new Date();
        this.isAccepted = false;
        this.shippingDate = new Date();
    }

}

File: synthese_frontend/src/components/Order/Order.js
import { Row, Col } from "react-bootstrap";

const Order = ({ order, onClick }) => {

  return (
    <Row className="list_node_item" onClick={() => onClick(order)}>
      <Col xs={9} className="list_node_text_item">
        <div>
          <li>
            {" "}
            {order.listingList.length} listing(s) - {order.cost} $, {order.address}, {order.city} {" "}
          </li>
        </div>
      </Col>
    </Row>
  );
};

export default Order;